partiql-eval/src/main/kotlin/org/partiql/eval/internal/operator/rex/ExprCallDynamic.kt

```diff
@@ -4,20 +4,30 @@ import org.partiql.errors.TypeCheckException
 import org.partiql.eval.internal.Environment
 import org.partiql.eval.internal.operator.Operator
 import org.partiql.eval.value.Datum
-import org.partiql.plan.Ref
 import org.partiql.spi.fn.Fn
 import org.partiql.types.PType
 import org.partiql.value.PartiQLValue
 import org.partiql.value.PartiQLValueExperimental
-import org.partiql.value.PartiQLValueType
 
 /**
  * This represents Dynamic Dispatch.
  *
  * For the purposes of efficiency, this implementation aims to reduce any re-execution of compiled arguments. It
- * does this by avoiding the compilation of [Candidate.fn] and [Candidate.coercions] into
+ * does this by avoiding the compilation of [Candidate.fn] into
  * [ExprCallStatic]'s. By doing this, this implementation can evaluate ([eval]) the input [Record], execute and gather the
- * arguments, and pass the [PartiQLValue]s directly to the [Candidate.eval].
+ * arguments, and pass the [PartiQLValue]s directly to the [Candidate.eval]. This implementation also caches previously
+ * resolved candidates.
+ *
+ * @param name the name of the function
+ * @param candidates the ordered set of applicable functions.
+ * @param args the arguments to the function
+ * @property paramIndices the last index of the [args]
+ * @property paramTypes represents a two dimensional array, where the first dimension maps to the [candidates], and the
+ * second dimensions maps to the [org.partiql.spi.fn.FnSignature.parameters]'s [PType].
+ * @property paramFamilies represents a two dimensional array, where the first dimension maps to the [candidates], and the
+ * second dimensions maps to the [org.partiql.spi.fn.FnSignature.parameters]'s [CoercionFamily]. This allows the algorithm
+ * to know if an argument is coercible to the target parameter.
+ * @property cachedMatches a memoization cache for the [match] function.
  */
 @OptIn(PartiQLValueExperimental::class)
 internal class ExprCallDynamic(
```

Comment:
In regards to all the private comments (knowing that this is internal).

Generally if expression operators need to share logic, then we can internalize that somewhere common rather than nested within a class.

However, I can see from partiql-eval.api that it doesn't matter — so the private comments are somewhat pointless.